### PR TITLE
18745 remove authorization from bivariate matrix endpoint

### PR DESCRIPTION
**Pull request by palina-krukovich**
## Summary by CodeRabbit

- **New Features**
  - Introduced user access control for viewing bivariate matrix data.

- **Refactor**
  - Removed `@PreAuthorize` annotation and integrated access control logic directly within the `getBivariateMatrix` method.

### CODE REVIEW

File: src/main/java/io/kontur/disasterninja/service/UserProfileService.java [hunk 1]
```diff
@@ -42,4 +46,21 @@ public UserDto updateUser(UserDto userDto) {
...
+     * Checks if user has access to any of the provided features
+     * @param featureNames feature names to check user access
+     * @return true if user has access to at least one of the provided features,
+     *         false otherwise
+     */
+    public boolean userHasAccessToFeature(List<String> featureNames) {
+        List<AppSummaryDto> appSummaries = client.getAppsList();
+        for (AppSummaryDto appSummary : appSummaries) {
+            AppDto app = client.getApp(appSummary.getId());
+            if (app.getFeatures().stream().anyMatch(feature -> featureNames.contains(feature.getName()))) {
```
Comment (gabriel-fallen) [inline src/main/java/io/kontur/disasterninja/service/UserProfileService.java:60]: Why don't we put everything in a single stream computation like
```java
client.getAppsList().stream().flatMap(summary -> client.getApp(summary.getId()).getFeatures().stream()).anyMatch(feature -> featureNames.contains(feature.getName()))
```
or something along those lines? 😅

**Review by gabriel-fallen — DISMISSED** [review]
LGTM 👍

File: src/main/java/io/kontur/disasterninja/service/UserProfileService.java (hunk 1 above)
Comment (palina-krukovich) [reply src/main/java/io/kontur/disasterninja/service/UserProfileService.java]: I am removing authorization at all for now. We need to investigate this more. Checking if feature is available for a user in any app, is too time consuming. We need to add appId parameter or think of a new way to check access.

**Review by gabriel-fallen — APPROVED** [review]
Lest we forget, we'll really need to put some sort of authorization back! 😅

**Pull request merged**